app/src/main/java/com/android/solvit/provider/ui/request/ListRequestsFeed.kt

```diff
@@ -247,16 +305,159 @@ fun ServiceRequestItem(request: ServiceRequest) {
+                      PaddingValues(top = 40.dp, start = 12.dp, end = 12.dp), // Increased padding
+              ) {
+                items(packages.size) { index ->
+                  // If package is selected, we display it bigger
+                  val isSelected = selectedIndex == index
+                  val size by
+                      animateDpAsState(
+                          targetValue = if (isSelected) 350.dp else 320.dp,
+                          label = "PackageCardSize")
+                  PackageCard(
+                      packageProposal = packages[index],
+                      isSelected = isSelected,
+                      modifier =
+                          Modifier.width(260.dp) // Slightly wider for better touch targets
+                              .height(size)
+                              .clickable { selectedIndex = if (isSelected) -1 else index }
+                              .testTag("PackageCard"),
+                      selectedPackage = selectedPackage)
+                }
+              }
+              Spacer(modifier = Modifier.height(16.dp)) // Add space between LazyRow and Button
+              Button(
+                  onClick = {
+                    selectedPackage.value?.let {
+                      requestViewModel.saveServiceRequest(
+                          request.copy(
+                              providerId = providerId,
+                              packageId = it.uid,
+                              status = ServiceRequestStatus.ACCEPTED))
+                      showDialog.value = false
+                    }
+                  },
+                  enabled = selectedPackage.value != null) {
+                    Text(text = "Propose Package")
+                  }
+            } else {
+              Card(
+                  modifier = Modifier.fillMaxWidth().padding(16.dp),
+                  elevation = CardDefaults.cardElevation(8.dp),
+                  colors = CardDefaults.cardColors(containerColor = colorScheme.primaryContainer)) {
+                    Column(
+                        modifier = Modifier.padding(16.dp).fillMaxWidth(),
+                        horizontalAlignment = Alignment.CenterHorizontally) {
+                          Text(
+                              text =
+                                  "No packages available. Please enter a price for this service:",
+                              fontSize = 18.sp,
+                              fontWeight = FontWeight.Bold,
+                              color = colorScheme.onPrimaryContainer)
+                          Spacer(modifier = Modifier.height(16.dp))
+                          var priceInput by remember { mutableStateOf("") }
+                          val isPriceValid = priceInput.matches(Regex("^[0-9]+(\\.[0-9]{0,2})?$"))
+                          TextField(
+                              value = priceInput,
+                              onValueChange = { input ->
```

Comment:
I'm trying to enter an invalid number (word for example) but I can't write it maybe we can add an invalidation input